### PR TITLE
Calibauc

### PR DESCRIPTION
**Pull request by ekellbuch**
This code includes a sample_weight variable to mask the inputs to CalibrationAUC.

### CODE REVIEW

File: robustness_metrics/metrics/uncertainty.py [hunk 1]
```diff
@@ -787,13 +802,34 @@ def update_state(self, y_true: Sequence[float], y_pred: Sequence[float],
...
     labels = _compute_correct_predictions(
         y_true, y_pred, dtype=self.dtype)
 
+    # use sample weight if provided
+    if sample_weight is not None:
+      sample_weight = tf.convert_to_tensor(sample_weight)
+      if tf.rank(sample_weight) != 1:
+        sample_weight = tf.reshape(sample_weight, [-1])
+      scores = tf.boolean_mask(scores, sample_weight)
+      labels = tf.boolean_mask(labels, sample_weight)
```
Comment (dusenberrymw) [inline robustness_metrics/metrics/uncertainty.py:831]: Same comment as in https://github.com/google-research/robustness_metrics/pull/77: if we add a `sample_weight` to the API here, we'd need to follow the tf.keras.metrics.Metric API and have `sample_weight` be an arbitrary _continuous_ tensor. Binary masking is a subset of possible case, and we'd need to think more about how to support that general case here. For binary masking, I'd suggest moving that to the calling site for now.

**Review by dusenberrymw — CHANGES_REQUESTED** [review]
Thanks, Kelly.
